ext/CEGUI/IrrlichtBaWTexture.cpp

```diff
@@ -148,17 +237,52 @@ void IrrlichtBaWTexture::loadFromMemory(const void* buffer, const CEGUI::Sizef&
     viewParams.subresourceRange.levelCount = 1;
     viewParams.viewType = decltype(viewParams.viewType)::ET_2D;
 
-    gpuTexture = driver->createGPUImageView(std::move(viewParams));
+    frameBuffer = createFrameBuffer(driver->createGPUImageView(std::move(viewParams)));
 }
 
 void IrrlichtBaWTexture::blitFromMemory(const void* sourceData, const CEGUI::Rectf& area)
 {
+    bool status = frameBuffer;
 
+    if (status)
+    {
+        auto gpuTexture = frameBuffer->getAttachment(textureColorAttachment);
+        auto& viewParams = gpuTexture->getCreationParameters();
+        auto& extent = viewParams.image->getCreationParameters().extent;
+        irr::core::recti coverArea(area.left(), area.top(), area.getWidth(), area.getHeight());
+
+        auto temporaryInFrameBuffer = createFrameBuffer(sourceData, viewParams.format, extent.width, extent.height);
+        driver->blitRenderTargets(temporaryInFrameBuffer, frameBuffer, false, false, coverArea, coverArea);
+        temporaryInFrameBuffer->drop();
+    }
+    else
+        assert(status);
```

Comment:
do not blit with fraebuffers!

you should be using `copyBufferToImage` here, and you could have implemented `IrrlichtBaWTexture::loadFromMemory` with this function partially

Reply:
adjusted in following commit

Reply:
but the area is covering area of framebuffer likely, how in that case I can copy it correctly?

Reply:
got an idea

Reply:
oh forget about regions haha